nengo_loihi/builder/ensemble.py

```diff
@@ -0,0 +1,190 @@
+        eval_points = eval_points.sample(n_points, ens.dimensions, rng)
+    else:
+        if (ens.n_eval_points is not None
+                and eval_points.shape[0] != ens.n_eval_points):
+            warnings.warn("Number of eval_points doesn't match "
+                          "n_eval_points. Ignoring n_eval_points.")
+        eval_points = np.array(eval_points, dtype=np.float64)
+        assert eval_points.ndim == 2
+
+    if scale_eval_points:
+        eval_points *= ens.radius  # scale by ensemble radius
+    return eval_points
+
+
+def get_gain_bias(ens, rng=np.random, intercept_limit=1.0):
+    if ens.gain is not None and ens.bias is not None:
+        gain = get_samples(ens.gain, ens.n_neurons, rng=rng)
+        bias = get_samples(ens.bias, ens.n_neurons, rng=rng)
+        max_rates, intercepts = ens.neuron_type.max_rates_intercepts(
+            gain, bias)
+    elif ens.gain is not None or ens.bias is not None:
+        # TODO: handle this instead of error
+        raise NotImplementedError("gain or bias set for %s, but not both. "
+                                  "Solving for one given the other is not "
+                                  "implemented yet." % ens)
+    else:
+        int_distorarray = ens.intercepts
+        if isinstance(int_distorarray, nengo.dists.Uniform):
+            if int_distorarray.high > intercept_limit:
+                warnings.warn(
+                    "Intercepts are larger than intercept limit (%g). "
+                    "High intercept values cause issues when discretizing "
+                    "the model for running on Loihi." % intercept_limit)
+                int_distorarray = nengo.dists.Uniform(
+                    min(int_distorarray.low, intercept_limit),
+                    min(int_distorarray.high, intercept_limit))
+
+        max_rates = get_samples(ens.max_rates, ens.n_neurons, rng=rng)
+        intercepts = get_samples(int_distorarray, ens.n_neurons, rng=rng)
+
+        if np.any(intercepts > intercept_limit):
+            intercepts[intercepts > intercept_limit] = intercept_limit
+            warnings.warn(
+                "Intercepts are larger than intercept limit (%g). "
+                "High intercept values cause issues when discretizing "
+                "the model for running on Loihi." % intercept_limit)
+
+        gain, bias = ens.neuron_type.gain_bias(max_rates, intercepts)
+        if gain is not None and (
+                not np.all(np.isfinite(gain)) or np.any(gain <= 0.)):
+            raise BuildError(
+                "The specified intercepts for %s lead to neurons with "
+                "negative or non-finite gain. Please adjust the intercepts so "
+                "that all gains are positive. For most neuron types (e.g., "
+                "LIF neurons) this is achieved by reducing the maximum "
+                "intercept value to below 1." % ens)
+
+    return gain, bias, max_rates, intercepts
+
+
+BuiltEnsemble = collections.namedtuple(
```

Comment:
Since #30, we have modified a few of those functions, mainly to use Loihi tuning curves for the activations when solving for decoders, I think. So I think that's why I skipped it. But many of them can probably be imported, so we should go through and check.